include/deal.II/matrix_free/matrix_free.h

```diff
@@ -3719,6 +3775,55 @@ namespace internal
         }
     }
 
+    void
+    reset_ghost_values(
+      const LinearAlgebra::SharedMPI::Vector<Number> &vec) const
+    {
+      return; // TODO: why do we need this function?
+              // I am reseting the ghost_values during compression
```

Comment:
I have not looked into all details, but generally `reset_ghost_values` is for the vector we called `update_ghost_values` on (but which we do not want to remain ghosted afterwards).